pymodbus/transaction.py

```diff
@@ -208,11 +201,11 @@ def _transact(self, packet, response_length, full=False):
                 _logger.debug("Changing transaction state from 'SENDING' "
                               "to 'WAITING FOR REPLY'")
                 self.client.state = ModbusTransactionState.WAITING_FOR_REPLY
-            result = self._recv(response_length or 1024, full)
+            result = self._recv(response_length, full)
```

Comment:
We can not pass `None` as response_length, in cases where a transaction returns no response, the argument `full` would be set to true and subsequent transactions would try to read the full length (`None` in some cases) throwing an error. Can we have 1024 back here ?

Reply:
I have made a wrong assumption, that is all messages have an expected_response_length.

If i get the 1024 back, the ```client._recv``` function will wait until it receives 1024 bytes or the timeout expires, so probably it always waits until the timeout expires, making the communication very slow.

Another solution should be to make the ```client._recv(size)``` to return the data as soon as at least 1 byte is received, without waits that timeout expires when ```size=None``` is passed.

What do you think about it?

Reply:
In most of the normal cases (read/write registers/coils) expected_length is always calculated so it should not be an issue, for some diagnostic or mei requests the response length is dynamic and hence we could choose to read 1024 bytes till time out ,which is still OK with me.